sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java

```diff
@@ -173,11 +170,11 @@ public FormRecognizerServiceVersion getServiceVersion() {
     }
 
     private Function<PollingContext<OperationResult>, Mono<OperationResult>> receiptStreamActivationOperation(
-        Flux<ByteBuffer> buffer, long length, FormContentType formContentType, boolean includeTextDetails) {
+        Flux<ByteBuffer> buffer, long length, boolean includeTextDetails) {
         return (pollingContext) -> {
             try {
                 return service.analyzeReceiptAsyncWithResponseAsync(includeTextDetails,
-                    ContentType.fromString(formContentType.toString()), buffer, length)
+                    getContentType(buffer.blockFirst()), buffer, length)
```

Comment:
We generally don't want to block in Reactive code as this could cause a RuntimeException for blocking in a non-blocking thread. We have some control on whether non-blocking threads are being used but not complete control.